File: VB/WindowsApplication54/DesignExtension.vb
Imports System
Imports System.Data
Imports System.Data.OleDb
Imports System.IO
Imports DevExpress.XtraReports.Extensions
Imports DevExpress.XtraReports.UI
Imports Microsoft.VisualBasic

Public Class DesignExtension
    Inherits ReportDesignExtension
    Protected Overrides Function CanSerialize(ByVal data As Object) As Boolean
        Return TypeOf data Is DataSet OrElse TypeOf data Is OleDbDataAdapter
    End Function
    Protected Overrides Function SerializeData(ByVal data As Object, ByVal report As XtraReport) As String
        If TypeOf data Is DataSet Then
            Return (TryCast(data, DataSet)).GetXmlSchema()
        End If
        If TypeOf data Is OleDbDataAdapter Then
            Dim adapter As OleDbDataAdapter = TryCast(data, OleDbDataAdapter)
            Return adapter.SelectCommand.Connection.ConnectionString & Constants.vbCrLf & adapter.SelectCommand.CommandText
        End If

        Return MyBase.SerializeData(data, report)
    End Function

    Protected Overrides Function CanDeserialize(ByVal value As String, ByVal typeName As String) As Boolean
        Return GetType(DataSet).FullName = typeName OrElse GetType(OleDbDataAdapter).FullName = typeName
    End Function
    Protected Overrides Function DeserializeData(ByVal value As String, ByVal typeName As String, ByVal report As XtraReport) As Object
        If GetType(DataSet).FullName = typeName Then
            Dim dataSet As New DataSet()
            dataSet.ReadXmlSchema(New StringReader(value))
            Return dataSet
        End If
        If GetType(OleDbDataAdapter).FullName = typeName Then
            Dim adapter As New OleDbDataAdapter()
            Dim values() As String = value.Split(Constants.vbCrLf.ToCharArray(), StringSplitOptions.RemoveEmptyEntries)
            adapter.SelectCommand = New OleDbCommand(values(1), New OleDbConnection(values(0)))
            Return adapter
        End If
        Return MyBase.DeserializeData(value, typeName, report)
    End Function
End Class

File: CS/WindowsApplication54/DesignExtension.cs
using System;
using System.Data;
using System.Data.OleDb;
using System.IO;
using DevExpress.XtraReports.Extensions;
using DevExpress.XtraReports.UI;

class DesignExtension : ReportDesignExtension {
    protected override bool CanSerialize(object data) {
        return data is DataSet || data is OleDbDataAdapter;
    }
    protected override string SerializeData(object data, XtraReport report) {
        if (data is DataSet)
            return (data as DataSet).GetXmlSchema();
        if (data is OleDbDataAdapter) {
            OleDbDataAdapter adapter = data as OleDbDataAdapter;
            return adapter.SelectCommand.Connection.ConnectionString +
                "\r\n" + adapter.SelectCommand.CommandText;
        }

        return base.SerializeData(data, report);
    }

    protected override bool CanDeserialize(string value, string typeName) {
        return typeof(DataSet).FullName ==
            typeName || typeof(OleDbDataAdapter).FullName == typeName;
    }
    protected override object DeserializeData(string value, string typeName, XtraReport report) {
        if (typeof(DataSet).FullName == typeName) {
            DataSet dataSet = new DataSet();
            dataSet.ReadXmlSchema(new StringReader(value));
            return dataSet;
        }
        if (typeof(OleDbDataAdapter).FullName == typeName) {
            OleDbDataAdapter adapter = new OleDbDataAdapter();
            string[] values = value.Split("\r\n".ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
            adapter.SelectCommand = new OleDbCommand(values[1], new OleDbConnection(values[0]));
            return adapter;
        }
        return base.DeserializeData(value, typeName, report);
    }
}